### ISSUE TITLE
How to download a link like this?

### ISSUE THREAD

**Issue by rfv1122** [aria2/aria2#1027]
Link form: http://qiannao.com/file/j00seng/e14c16ff/
Download link is: http://dl51.qiannao.com/downfile/j00seng/e14c16ff?md5=2311bcd74e220d9c35bb81e80e098259
I tried to give referer, header, but I don't know where am I wrong, I can't download file completely. How to download a link like this?
I don't know where to ask, or maybe it's a bug? Please somebody help, thx. Also sorry, if I can't ask this question here.

**Comment by SlyzertVoltrond** [comment]
You need to pass along your web browser's cookies file: https://aria2.github.io/manual/en/html/aria2c.html#cmdoption--load-cookies

Mine, for example, is located at .mozilla/firefox/s36sbfl7.default/cookies.sqlite (Firefox on Linux)

**Comment by rfv1122** [comment]
Thank you very much, but I gave the cookie you said in my folder to aria2c, but it still doesn't download file success. Maybe there is another parameter need in that link. My OS is Win 10. aria2c 1.32.

**Comment by SlyzertVoltrond** [comment]
Yes, I'm able to download that file by supplying --load-cookies. What web browser are you using? If it's Mozilla Firefox, see this thread to find the cookies file: https://support.mozilla.org/en-US/questions/1096646

**Comment by rfv1122** [comment]
I already loaded the correct cookies.sqlite just after your previous post said. I am very sure it loaded. I am also able to download file, but the file after downloaded is only 20KB, not 48.31 MB, I also tried install fresh Firefox, download file is not correct, too. I think I miss some parameters.

**Comment by SlyzertVoltrond** [comment]
Hmm, not sure what's keeping aria2 from fully downloading the file. Maybe try downloading it partially through Firefox first, cancel it, and then download the link using aria2 with --load-cookies option.

What's the error message when aria2 quits with only 20KB downloaded?

**Comment by rfv1122** [comment]
Thanks for you always replying me, there is no error message shows. I fonund why can't I use `--load-cookies` [here](https://github.com/aria2/aria2/issues/845#issuecomment-276655877).
But I still encounter problem to use header, it always invalid. My RPC script is: http://jsoneditoronline.org/?id=4ee8fb1e0314e124bd3ab7d4b2ed19f1

**Comment by rfv1122** [comment]
I use this command, download is also failed: `aria2c http://dl51.qiannao.com/downfile/j00seng/e14c16ff?md5=2311bcd74e220d9c35bb81e80e098259 --header="language: en_us" --header="validCodeUrl: http://qiannao.com:9090/view?module=service&action=queryValidCode" --header="vid: 5337fb82" --header="vid1: 71737c92a63170ee" --log-level=debug --log=aria2.log`
This is [aria2.log](https://github.com/aria2/aria2/files/1386090/aria2.log). `vid1` is changing, I get it after normal browser download (download by browser is no problem).
OS: Win 10 64bit, aria2 1.32.0 64bit.

**Comment by rfv1122** [comment]
Would somebody tell me this problem cause by me or it's something like a bug?

**Comment by 102286545** [comment]
The header param is a string,change the request params.You can success:
{
  "jsonrpc": "2.0",
  "id": "123",
  "method": "aria2.addUri",
  "params": [
    [
      "http://dl51.qiannao.com/downfile/j00seng/e14c16ff?md5=2311bcd74e220d9c35bb81e80e098259# "
    ],
    {
      "header":"language:en_us\nvalidCodeUrl:http://qiannao.com:9090/view?module=service&action=queryValidCode\nvid:5337fb82\nvid1:71737c92a78d326a"
    }
  ]
}
